pkg/orchestration/wiring/edge/types.go

```diff
@@ -2,15 +2,24 @@ package edge
 
 import "github.com/atlassian/voyager"
 
-type Parameters struct {
-	UpstreamAddress `json:"upstream_address"`
-	UpstreamPort    int      `json:"upstream_port,omitempty"`
-	UpstreamSuffix  string   `json:"upstream_suffix,omitempty"`
-	UpstreamOnly    string   `json:"upstream_only,omitempty"`
-	Domain          []string `json:"domain,omitempty"`
-	Healthcheck     string   `json:"healthcheck,omitempty"`
-	Rewrite         string   `json:"rewrite,omitempty"`
-	Routes          `json:"routes"`
+type InstanceParameters struct {
+	ServiceName string             `json:"serviceName"`
+	Resource    ResourceParameters `json:"resource"`
+}
+
+type ResourceParameters struct {
+	Attributes Attributes `json:"attributes"`
+}
+
+type Attributes struct {
+	UpstreamAddress *UpstreamAddress `json:"upstream_address"`
+	UpstreamPort    int              `json:"upstream_port,omitempty"`
+	UpstreamSuffix  string           `json:"upstream_suffix,omitempty"`
+	UpstreamOnly    string           `json:"upstream_only,omitempty"`
+	Domain          []string         `json:"domain,omitempty"`
+	Healthcheck     string           `json:"healthcheck,omitempty"`
+	Rewrite         string           `json:"rewrite,omitempty"`
+	Routes          Routes           `json:"routes,omitempty"`
 }
 
 type UpstreamAddress []struct {
```

Comment:
OOC: What does this mean? I.e. how does this type work.
```
x := UpstreamAddress{}
x[0] = ???
````

Reply:
Interesting, never saw this pattern before :)

@roaanv type is just an alias for a particular structure, e.g. you can define `type Bla string` which will operate the same way as a normal string. Here a type is defined as "slice of anonymous structs".
> x := UpstreamAddress{}

it's the same as
> x := []struct{}

i.e. it's an empty slice and `x[0]` will panic, see https://play.golang.org/p/mkYe8VkxR5i

Reply:
```suggestion
 type UpstreamAddress struct {
```

Reply:
Givent that the struct is anonymous, appending items to this slice is very ugly (but works!) https://play.golang.org/p/DDLXHxTKqT8